packages/sdk/src/mb-impl/token.ts

```diff
@@ -0,0 +1,214 @@
+// Mintbase token contract JS implementation
+
+import { TransactionArgs } from '../calls';
+import {
+  DEFAULT_MB_LOGO,
+  METHOD_NAMES,
+  MB_MAINNET_TOKEN_FACTORY,
+  MB_TESTNET_TOKEN_FACTORY,
+  Network,
+} from './constants';
+import {
+  TransferArgs,
+  BurnArgs,
+  SingleTokenArgs,
+  DeployTokenContractArgs,
+  AccountId,
+  TransferTokenContractOwnership,
+  MintArgs,
+  AddRemoveMinterArgs,
+  BatchChangeMinters,
+  RevokeAccountArgs,
+} from './token.types';
+
+// TODO: how to deal with this differently?
+const getFactoryContract = (network: string): AccountId => {
+  if (network === Network.MAINNET) return MB_MAINNET_TOKEN_FACTORY;
+  else return MB_TESTNET_TOKEN_FACTORY;
+};
+
+// TODO: figure out a way to generate gas and deposit for each
+
+export const transfer = (args: TransferArgs): TransactionArgs => {
+  const { nftContractId, transfer } = args;
+
+  const isBatchTransfer = transfer.length > 1;
+
+  if (isBatchTransfer) {
+    const ids = transfer.map((transferElm) => {
+      return [transferElm.receiverId, transferElm.tokenId];
+    });
+
+    return {
+      contractAddress: nftContractId,
+      methodName: METHOD_NAMES.BATCH_TRANSFER,
+      args: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        token_ids: ids,
+      },
+    };
+  } else {
+    const receiverId = transfer[0].receiverId;
+    const tokenId = transfer[0].tokenId;
+
+    return {
+      contractAddress: nftContractId,
+      methodName: METHOD_NAMES.TRANSFER,
+      args: {
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        receiver_id: receiverId,
+        // eslint-disable-next-line @typescript-eslint/camelcase
+        token_id: tokenId,
+      },
+    };
+  }
+};
+
+export const burn = (args: BurnArgs): TransactionArgs => {
+  const { nftContractId, tokenIds } = args;
+
+  return {
+    contractAddress: nftContractId,
+    methodName: METHOD_NAMES.BATCH_BURN,
+    args: {
+      // eslint-disable-next-line @typescript-eslint/camelcase
+      token_ids: tokenIds,
+    },
+  };
+};
+
+export const deployContract = (
+  args: DeployTokenContractArgs,
+): TransactionArgs => {
+  const {
+    name,
+    factoryContractId,
+    network = Network.TESTNET,
+    ownerId,
+    metadata,
+  } = args;
+
+  const { symbol, icon, baseUri, reference, referenceHash } = metadata;
+
+  const data = {
+    // eslint-disable-next-line @typescript-eslint/camelcase
+    owner_id: ownerId,
+    metadata: {
+      spec: 'nft-1.0.0',
+      name: name.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
+      symbol: symbol.replace(/[^a-z0-9]+/gim, '').toLowerCase(),
```

Comment:
Same here about returning an error instead of overwriting. Also currently a lot of symbols are all upper case, so why force them to be lower case now?